Implement Flask-Login's alternate_id with fs_uniquifier

Kishi85 opened the pull request:
Fixes #239

jwag956 commented on flask_security/core.py:
```diff
@@ -343,10 +343,21 @@
-    return user
+    # Try to load based on fs_uniquifier (alternative_id) first
+    try:
+        user = _security.datastore.find_user(fs_uniquifier=user_id)
```
Since most users of FS don't yet have fs_uniquifier - lets be nice and not potentially do multiple DB calls by doing a simple check that the user model has such an attribute first.

Kishi85 commented:
> unit test to show 'old' non-fs_uniqufier still works?
This is done by any test already is it not? fs_uniquifier is only touched in test_change_uniquifier?

> Doc string for get_id so is shows up in our API documentation.
> 
Added.

> we need to document somewhere about this so folks know we are using fs_uniquifier - possibly just a sentance or 2 in the features.rst where it talks about alternative_token?
>
I've added a small note to the session based authentication topic in features.rst

> Add to docstring of set_uniquifier to add that it will render remember cookies and session cookies invalid as well.
Added.

I've also added a check for existance and validity of fs_uniquifier in the respective changes.

jwag956 commented on docs/features.rst:
```diff
@@ -12,10 +12,11 @@ Session based authentication is fulfilled entirely by the `Flask-Login`_
-CSRF support. Flask-Security extends that to support requiring CSRF for
-requests that are authenticated via session cookies, but not for requests
-authenticated using tokens.
+expired. Flask-Security uses ``fs_uniquifier`` from it's Token Authentication
```
nit: 'its'

Kishi85 replied:
fixed
